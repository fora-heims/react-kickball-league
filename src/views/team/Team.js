import { React, useState, useEffect } from 'react';
import './Team.css';
import TeamDetail from '../../components/teamdetail/TeamDetail.js';
import { getTeam } from '../../services/teams';

export default function Team(props) {
  const [team, setTeam] = useState([]);
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;
  useEffect(() => {
    const fetchTeam = async () => {
      const teamData = await getTeam(id);
      setTeam(teamData);
      setLoading(false);
    };
    fetchTeam();
  }, [id]);
  if (loading) {
    return <div>loading</div>;
  }
  return (
    <div>
      <TeamDetail team={team[0]} showDetail></TeamDetail>
    </div>
  );
}
