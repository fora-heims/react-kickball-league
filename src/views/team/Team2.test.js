import Team from './Team.js';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

it('should render a teams details', async () => {
  const { container } = render(
    <MemoryRouter>
      <Team match={{ params: { id: 2 } }} />
    </MemoryRouter>
  );
  await screen.findByText('Stumptown Lumberjacks');
  expect(container).toMatchSnapshot();
});
