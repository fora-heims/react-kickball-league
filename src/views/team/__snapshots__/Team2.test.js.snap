// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a teams details 1`] = `
<div>
  <div>
    <div>
      <h3>
        Stumptown Lumberjacks
      </h3>
      <h4>
        Location: 
        Portland
        , 
        OR
      </h4>
      <h4>
        Players:
      </h4>
      <ul>
        <li>
          <a
            href="/players/5"
          >
            Gary Coolman
          </a>
          <h5>
            Position: 
            Pitcher
          </h5>
        </li>
        <li>
          <a
            href="/players/6"
          >
            Terry Cruise
          </a>
          <h5>
            Position: 
            Catcher
          </h5>
        </li>
        <li>
          <a
            href="/players/7"
          >
            Bud E. Guy
          </a>
          <h5>
            Position: 
            Outfield
          </h5>
        </li>
        <li>
          <a
            href="/players/8"
          >
            Buster Rimes
          </a>
          <h5>
            Position: 
            Shortstop
          </h5>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
